webpack/components/extensions/HostDetails/DetailsTabCards/RegistrationCard.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { translate as __ } from 'foremanReact/common/I18n';
+import {
+  DescriptionList,
+  DescriptionListTerm,
+  DescriptionListGroup,
+  DescriptionListDescription,
+  Text,
+  TextVariants,
+} from '@patternfly/react-core';
+import { urlBuilder } from 'foremanReact/common/urlHelpers';
+import CardTemplate from 'foremanReact/components/HostDetails/Templates/CardItem/CardTemplate';
+
+export const RegisteredBy = ({ user, activationKey }) => {
+  const activationKeyID = activationKey?.id;
+  const activationKeyName = activationKey?.name;
+  if (user) {
+    return (
+      <DescriptionListDescription>{user}</DescriptionListDescription>
+    );
+  }
+  return (
+    <>
+      <Text component={TextVariants.h4}>Activation key</Text>
+      <a href={urlBuilder(`activation_keys/${activationKeyID}`, '')}>{activationKeyName}</a>
+    </>
+  );
+};
+
+RegisteredBy.propTypes = {
+  user: PropTypes.string,
+  activationKey: PropTypes.object, // eslint-disable-line react/forbid-prop-types
+};
+
+RegisteredBy.defaultProps = {
+  user: '',
+  activationKey: {},
+};
+
+const RegistrationCard = ({ isExpandedGlobal, hostDetails }) => {
+  const registerDate = hostDetails?.subscription_facet_attributes?.registered_at;
+  const registerEndpoint = hostDetails?.subscription_facet_attributes?.registered_through;
+  const registerUser = hostDetails?.subscription_facet_attributes?.user?.login;
+  const registerActivationKey = hostDetails?.subscription_facet_attributes?.activation_keys[0];
```

Comment:
`propsToCamelCase` can help a lot here:

```js
import { propsToCamelCase } from 'foremanReact/common/helpers';
...
const subscriptionFacetAttributes = propsToCamelCase(hostDetails?.subscription_facet_attributes);
const registerDate = subscriptionFacetAttributes?.registeredAt;
const registerEndpoint = subscriptionFacetAttributes?.registeredThrough;
const registerUser = subscriptionFacetAttributes?.user?.login;
const registerActivationKey = subscriptionFacetAttributes?.activationKeys[0];
```

and notice we haven't even done any destructuring yet! But hopefully you're beginning to see how we get there..